### ISSUE TITLE
onMouseUp event handler not being called on paths below other paths

### ISSUE THREAD

**Issue by samueller** [paperjs/paper.js#372]
If I have:
```
path.onMouseUp = function(event) {
  alert('hello');
};
```
then I'll get an alert with hello as expected when I unclick the mouse on top of that path. However, if I have another path object on top of it, then the alert never gets shown. How do I make mouseUp events propagate down through all the objects?

**Comment by chad-autry** [comment]
My current experience is that onMouseDown (and I suspect all other item level mouse interactions) is also blocked by other items higher in the z-index order.

Use Case 1: I've rasterized a grid pattern, for performance, after building it from path items. Once transformed into a raster; the mouseDown listener for items seen through the transparent areas is blocked.

Use Case 2: I have translucent shapes to make a lens flare effect. These also block mouse interaction.

These both might be on your list of changes to make in the refactor mentioned in https://github.com/paperjs/paper.js/issues/536 and could be solved by either propagating mouse events all the way down, or allowing items to be flagged as ignoring mouse events.

Use case 1 (rasterizing complex objects for performance) is also solvable individually by checking the alpha of the raster in _hitTestSelf. Perhaps gated by a flag, defaulted to current functionality. Would you appreciate a pull request for the Raster portion of the problem?

**Comment by lehni** [comment]
I am closing this since #842 tracks the same issue.

**Comment by lehni** [comment]
Actually, this wasn't quite the same issue, but: You wouldn't get this behavior in HTML either, and I think it's actually correct for it to only be called on the item that you actually visually click on, and from there to bubble up the hierarchy of its parents.

**Comment by chad-autry** [comment]
I actually had a fix for clicking through a transparent section of an
image. For my own use case, however, I started handling logical layers more
in my own code so I no longer needed it.

Would you like me to get the change out for you to look at?
